### PR TITLE
N12 mutex unsafe design

### PR DESCRIPTION
**Pull request by levBagryansky**
@yegor256 please review
There will be 2 new atoms: `thread.acquire.unsafe-acquire`  and `thread.lock.release.unsafe-release`. Mechanism of `thread.acquire.@` and `thread.lock.release.@` will be implement by eo.

### CODE REVIEW

File: src/main/eo/org/eolang/threads/mutex.eo [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
++architect yegor256@gmail.com
++home https://github.com/objectionary/eo
++package org.eolang.threads
++rt jvm org.eolang:eo-runtime:0.0.0
```
Comment (yegor256) [inline src/main/eo/org/eolang/threads/mutex.eo:26]: @levBagryansky you should say `eo-threads` here instead

File: src/main/eo/org/eolang/threads/mutex.eo [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
++architect yegor256@gmail.com
++home https://github.com/objectionary/eo
```
Comment (yegor256) [inline src/main/eo/org/eolang/threads/mutex.eo:24]: @levBagryansky also here, the `home` is set wrong

File: src/main/eo/org/eolang/threads/mutex.eo [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
++architect yegor256@gmail.com
```
Comment (yegor256) [inline src/main/eo/org/eolang/threads/mutex.eo:23]: @levBagryansky also this info is not valid

File: src/main/java/EOorg/EOeolang/EOthreads/Semaphores.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        // Singleton
+    }
+
+    /**
+     * Get.
+     *
+     * @param mutex Phi mutex as a key
+     * @return Semaphore matching to the key
+     */
+    public Semaphore get(final Phi mutex) {
```
Comment (yegor256) [inline src/main/java/EOorg/EOeolang/EOthreads/Semaphores.java:70]: @levBagryansky it seems that passing `permits` to this method would be a cleaner design

**Comment by yegor256** [comment]
@mximp please review

File: src/main/java/EOorg/EOeolang/EOthreads/Semaphores.java (hunk 4 above)
Comment (levBagryansky) [reply src/main/java/EOorg/EOeolang/EOthreads/Semaphores.java:70]: Then we will initialize `permits` twice: in `unsafe_acquire` and `unsafe_release`. We also does not need permits if the semaphore is already exist in `HashMap`.

File: src/main/eo/org/eolang/threads/mutex.eo (hunk 2 above)
Comment (levBagryansky) [reply src/main/eo/org/eolang/threads/mutex.eo]: ok

File: src/main/eo/org/eolang/threads/mutex.eo (hunk 1 above)
Comment (levBagryansky) [reply src/main/eo/org/eolang/threads/mutex.eo:26]: ok

**Comment by levBagryansky** [comment]
@Graur please review

File: src/main/eo/org/eolang/threads/sleep.eo [hunk 5]
```diff
@@ -20,8 +20,8 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-+architect yegor256@gmail.com
-+home https://github.com/objectionary/eo
++architect 28lev11@gmail.com
++home https://github.com/objectionary/eo-threads
 +package org.eolang.threads
 +rt jvm org.eolang:eo-runtime:0.0.0
```
Comment (Graur) [inline src/main/eo/org/eolang/threads/sleep.eo:26]: @levBagryansky Same here, maybe it should be `eo-thread`?

**Review by Graur — CHANGES_REQUESTED** [review]
@levBagryansky Looks good to me, just a few simple comments below:

File: src/main/eo/org/eolang/threads/sleep.eo (hunk 5 above)
Comment (levBagryansky) [reply src/main/eo/org/eolang/threads/sleep.eo:26]: @Graur name of repository is eo-threads

File: src/main/eo/org/eolang/threads/mutex.eo [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
++home https://github.com/objectionary/eo-threads
++package org.eolang.threads
++rt jvm org.eolang:eo-runtime:0.0.0
++version 0.0.0
+
+[permits] > mutex
+  # gives a permit and return personal lock
+
+  # @todo #12:90min Make it possible to acquire
+  #  the mutex in sec. Now it requires the
```
[the remark this answers is not in the document]
Comment (levBagryansky) [reply src/main/eo/org/eolang/threads/mutex.eo:33]: yes

File: src/main/eo/org/eolang/threads/sleep.eo (hunk 5 above)
Comment (Graur) [reply src/main/eo/org/eolang/threads/sleep.eo:26]: @levBagryansky But still not eo-runtime, right?

**Review by Graur — APPROVED** [review]
@levBagryansky Looks good to me

**Comment by levBagryansky** [comment]
@yegor256 please have a look

**Comment by levBagryansky** [comment]
@yegor256 ok

**Pull request closed without merging**